README.markdown

```diff
@@ -99,22 +102,62 @@ To build from source, do:
+
+For those looking to extend Cappuccino for touch applications, one option worth exploring is the open source Objective-C framework ["Chameleon"](https://github.com/BigZaphod/Chameleon). This predates Apple's Catalyst efforts and ports many UIKit classes to AppKit, allowing for a seamless integration of touch features. See below for porting Objective-C code to Objective-J, something which is mostly a search-and-replace operation.
+
+**Q: How easy is it to port Objective-C code to Objective-J?**  
+**A:** Porting Objective-C code to Objective-J is mostly a search-and-replace operation, primarily focusing on adapting pointer syntax and class names to fit the Objective-J framework.  
+
+Here is a summary of the process:  
+
+1. **Replace class prefixes**: Objective-C classes with the 'NS' prefix (from the Cocoa framework) need to be replaced with the 'CP' prefix, which is used by Cappuccino. Objective-C does not support formal namespaces - class prefixes are a convention used universally to overcome this ('NS' for Apple/NeXTSTEP, 'CP' for Cappuccino). Third-party code may use another class prefix - or it may use none. For example:  
+
+    `NSString` becomes `CPString`,
+`NSArray` becomes `CPArray`, and
+`NSDictionary` becomes `CPDictionary`
+2. **Remove pointer references:** Objective-C uses pointers, indicated by the * symbol, which are not used in Objective-J. Thus, pointer references need to be removed. For instance:
+
+    `NSString *string = @"Hello";` becomes `CPString string = @"Hello";`
+
+3. **Adapt memory management:** Objective-C's manual memory management and automatic reference counting (ARC) do not apply to Objective-J. Thus, methods like retain, release, and autorelease are removed. Objective-J relies on JavaScript's garbage collection - interaction with it is not supported by any browser.
+
+5. **Modify framework-specific code:** Any code that directly interacts with Cocoa frameworks may need to be rewritten to fit the Cappuccino framework's equivalent.  Although almost always identical, this is not always true. This involves not only class replacements but also adapting any framework-specific functionality to Objective-J's counterparts.  
+
+6. **Modify code with no direct equivalent in Objective-J:** The most common of these is usage of the C language `enum` construct. Javascript and Objective-J provide no direct equivalent - it will usually be necessary to rewrite using Javascript objects or functions. 
+
+    #### Example Conversion
+Here’s a simple example of porting Objective-C code to Objective-J:
+
+    *Objective-C:*     
+```NSString *greeting = [[NSString alloc] initWithFormat:@"Hello, %@!", name];
+[greeting release];```. 
+ 
+    *Objective-J:*  
+```var greeting = CPString.stringWithFormat("Hello, %@!", name);```   
```

Comment:
shouldn't this read var greeting = [CPString stringWithFormat:@"Hello, %@!", name];

Reply:
Indeed, that should be `var greeting = [CPString stringWithFormat:@"Hello, %@!", name];`.
By why not generally go to more modern JavaScript, e.g.
`let greeting = [CPString stringWithFormat:@"Hello, %@!", name];` (or `const`).